scripts/generate-CHANGES.py

```diff
@@ -0,0 +1,320 @@
+import subprocess
+import sys
+import re
+
+def textwidth(text):
+	'''
+		Calculate the with of 'text' in number
+		of characters.
+
+		This function counts tabs as 8 characters.
+	'''
+	count = 0
+	for char in text:
+		if char == '\t':
+			count += 8
+		else:
+			count += 1
+	return count
+
+
+def perlCompatibleTextWrap(text, width, initial_indent, subsequent_indent):
+	'''
+		Format text like Perl's Text::Wrap#wrap function.
+
+		We do this to keep compatibility with existing formatting in
+		the CHANGES file. (Invoking this script overwrites the old file,
+		so the output of this function must match the Perl function's
+		output.)
+	'''
+
+	lines = []
+	line = ""
+
+	# Require at least one empty character
+	# of free space at the end of the line.
+	width -= 1
+
+	# Create our word queue by splitting the commit
+	# description by a regular space character (' ').
+	#
+	# This is a little simplistic, but it works in practice.
+	# In an ideal world, we should probably split all
+	# whitespace, so our queue guaranteed to only contain
+	# words.
+	queue = text.split(' ')
+
+	# Iterate through each word and in the queue
+	# and construct our formatted text.
+	while len(queue) > 0:
+		word = queue.pop(0)
+
+		# Which indentation mode should we use for
+		# this line?
+		# If this is the first line, use 'initial_indent'.
+		# If not, use 'subsequent_indent'.
+		if len(lines) == 0:
+			 indent = initial_indent
+		else:
+			indent = subsequent_indent
+
+		# If nothing has been added to our line yet,
+		# add the indentation.
+		if line == "":
+			line = indent
+
+		# Find the wordPrefix.
+		#
+		# The wordPrefix represents
+		# the string that will comes before
+		# 'word' when 'word' is added to the line.
+		#
+		# If no other words have been added to the
+		# line, the wordPrefix is the indentation.
+		# (Either initial_indnet or subsequent_indent).
+		# If other words have been added to the line
+		# already, the wordPrefix will be the existing
+		# content of the line, plus a space.
+		if line == indent:
+			wordPrefix = line
+		else:
+			wordPrefix = line + " "
+
+		# Calculate the width of line with the word added.
+		widthWithWordAdded = textwidth(wordPrefix + word)
+
+		# Check if the line would become too long if the word
+		# was added to the line as-is (considering the wordPrefix).
+		if widthWithWordAdded > width:
+			# Handle words that need to be split.
+			# As a rule, split words MUST start at the very beginning
+			# of a line (considering indentation) -- that is, there
+			# most be no other words in the line for a split word to
+			# be output.
+			#
+			# We check:
+			# (1) If the word is too long to be the only word on a line
+			# with the current width and indentation settings, and...
+			# (2) That the line only contains indentation.
+			#     This is the same as saying that the current line has
+			#     no other words in it.
+			#
+			# Both (1) and (2) must be true for the split word to be
```

Comment:
I believe that the old condition of `textwidth(indent + word)` was always true, because:

`widthWithWordAdded > width` expands to either `textwidth(indent + word) > width` (same as the old (1) condition) or `textwidth(line + " " + word) > width` (longer than (1)).

I have simplified the code.  Thanks!

Also, I can't see where the length of the indentation blocks matter?